### PR TITLE
[incubator/ado-agent] Added helm chart for azure devops build agent

### PR DESCRIPTION
**Pull request by gambtho**
#### What this PR does / why we need it:

This chart provides a self-hosted linux agent for use with Azure DevOps.   It is not limited to Azure Kubernetes Service, but can be run anywhere that allows outbound connection.   These agents provide additional workers that can be customized for use during an Azure DevOps pipeline

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X ] Chart Version bumped
- [X ] Variables are documented in the README.md
- [ X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by maorfr** [comment]
please take a look at helm/helm#6095 (comment)

we strongly encourage people to self host chart repos:
https://github.com/helm/hub/blob/master/Repositories.md

you can even do that using GitHub pages:
https://github.com/helm/hub/blob/master/docs/github.md#github-pages-as-a-repository

File: incubator/ado-agent/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+appVersion: "1.0"
```
Comment (mattfarina) [inline incubator/ado-agent/Chart.yaml:2]: I see the repo is tagged with the version 1.1. You might want to update the `appVersion` to the latest tag.

File: incubator/ado-agent/templates/secret.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: vsts
+type: Opaque
+data:
+  account: {{ .Values.vsts.account }}
+  token: {{ .Values.vsts.token }}
+  pool: {{ .Values.vsts.pool }}
```
Comment (mattfarina) [inline incubator/ado-agent/templates/secret.yaml:9]: The secrets need to be base64 encoded. You can see examples of doing this in charts at https://github.com/helm/charts/search?q=b64enc&unscoped_q=b64enc

File: incubator/ado-agent/values.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  pullPolicy: Always
+ingress:
+  enabled: false
+vsts:
+  account: "replacme"
+  token: "replaceme"
+  pool: "replaceme"
+service:
+  type: ClusterIP
+resources:
```
Comment (mattfarina) [inline incubator/ado-agent/values.yaml:14]: Can you please comment out the resources. There are details in the review guidelines about it at https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#configuration

**Comment by gambtho** [comment]
self hosting -- per comments above

**Pull request closed without merging**